package.json

```diff
@@ -236,6 +242,7 @@
     "tape": "git+https://github.com/kgryte/tape.git#fix/globby",
     "to-vfile": "^6.0.0",
     "typedoc": "^0.16.11",
+    "typescript": "4.3.5",
```

Comment:
Why are we dependent on an old version of TypeScript? I believe the latest stable is v5.2.2.

Also, what does this mean for us going forward? Does this mean we'll only be linting against a single TypeScript version? One of the benefits of `dtslint` is that our definitions were linted across multiple TypeScript versions. If this is not supported by the current migration, what are our plans for replicating this?

Reply:
@kgryte I picked the oldest TS version that was compatible with our dependencies given that we promise backwards compatibility to TS `4.1`. Can change to the latest version here.

Reply:
That makes sense. Would be good to document that choice and rationale in the OP.

That still leaves the question of how we can test across multiple TS versions as was the case with `dtslint`. TS doesn't have the greatest track record with semver, IIRC, so testing across versions is useful. May not be required for local development when, e.g., committing, but we likely do want tooling to support this (i.e., Make), and have an associated CI job which does more comprehensive testing.

This can be a follow-up PR, but I do think it is something that we should do, similar to how we run unit tests on Node.js (with tooling to run across multiple versions) and run across multiple versions on CI.

Reply:
This was documented in the OP.